docs/guide/snaps-introduction.md

```diff
@@ -0,0 +1,69 @@
+# Snaps Tutorial
+
+## What is Snaps?
+
+Snaps are a new evolution of MetaMask's functionality. By using MetaMask Flask in conjunction with the Snaps API, developers are able to bring their own code to the wallet in a custom-made code laboratory. MetaMask is the first wallet to provide this openness and flexibility to developers, paving the way to scale the number of features and personalization available to the wallet and your Dapp.
+
+Snaps allows developers to build and plug in their own APIs to MetaMask. No longer do you have to rely on MetaMask to roll out features and updates. The rate of innovation, and MetaMask's flexibility, are limited only by your imagination and your capacity to build. 
```

Comment:
The first sentence is good except it kinda sells snaps short a bit. Building APIs doesn't quite encompass things like enhancements to the wallet itself or integrating with internal MetaMask APIs.

The part about not having to rely on MetaMask to roll out new features and updates doesn't ring true to me. I get what you were going for, that one of the goals of snaps is to allow developers to innovate faster than our team alone could hope to. But maybe we can phrase this more carefully. Snap developers will still very much be dependent upon us and our release schedule.

The last sentence doesn't ring true either :sweat_smile: There remain MANY limitations to what developers can do with snaps. They run in a very restricted environment and the API they are given is quite small right now. Frankly it's probably the most restrictive environment most snap devs will have ever encountered.

Reply:
-I wasn't sure how much of that we were going to push, and how soon; hence I left it deliberately vague
-There was copy you found elsewhere--about how we'll use Flask to get a better idea of what devs want, and incorporate those things faster--that might resolve your doubts here.
-How about we nix that last sentence, then?

Reply:
Sorry for the fractured conversation - was thinking that [the draft here](https://github.com/MetaMask/metamask-docs/pull/331#discussion_r786326845) could replace both of these paragraphs. Do you think we need anything from this one?

I feel like there is still _something_ to say about the rate of innovation but I couldn't think of how to say it. Maybe the Flask description gets close to it. It's not an exact parallel because Flask is basically just for allowing experimentation and gaining feedback to improve the snaps system, whereas with snaps the ultimate goal isn't for all of the good ideas to get absorbed into MetaMask itself. The good ideas can stay as snaps, that's fine. But there is a parallel in that the systems are both meant to drive innovation, and they work together toward that goal.

But I don't know how we should say any of that here, and it seems fine if we omit it for now.